packages/webview_flutter/webview_flutter/android/src/main/java/io/flutter/plugins/webviewflutter/FlutterWebView.java

```diff
@@ -294,6 +312,65 @@ private void loadUrl(MethodCall methodCall, Result result) {
     result.success(null);
   }
 
+  private void loadRequest(MethodCall methodCall, final Result result) {
+    final WebViewRequest webViewRequest = buildWebViewRequest(methodCall);
+    if (webViewRequest == null) {
+      result.error("missing_args", "Missing arguments", null);
+    } else {
+      switch (webViewRequest.getMethod()) {
+        case GET:
+          webView.loadUrl(webViewRequest.getUrl(), webViewRequest.getHeaders());
+          break;
+        case POST:
+          if (webViewRequest.getHeaders().isEmpty()) {
+            webView.postUrl(webViewRequest.getUrl(), webViewRequest.getBody());
+          } else {
+            // Execute the request manually to be able to provide headers with the post request.
+            httpRequestManager.requestAsync(
+                webViewRequest,
+                new HttpRequestCallback() {
+                  @Override
+                  public void onComplete(String content) {
+                    if (!webView.isAttachedToWindow()) {
+                      result.error(
+                          "webview_destroyed",
+                          "Could not complete the post request because the webview is destroyed",
+                          null);
+                    } else {
+                      webView.loadDataWithBaseURL(
+                          webViewRequest.getUrl(), content, "text/html", "UTF-8", null);
```

Comment:
Are we certain that the request URL is always correct? E.g., what happens here in the case of a server-side redirect to a different origin?